generator/src/test/java/com/scottlogic/deg/generator/walker/reductive/ReductiveDecisionTreeReducerTests.java

```diff
@@ -323,6 +323,124 @@ public void shouldReturnNullIfALowLevelDecisionIsInvalidated(){
+
+        Assert.assertThat(reduced.getAtomicConstraints(), containsInAnyOrder(isEqualTo1));
+        Assert.assertThat(reduced.getDecisions().size(), equalTo(1));
+        Assert.assertThat(
+            AtomicConstraintExtractor.getAllAtomicConstraintsRecursively(reduced),
+            containsInAnyOrder(isEqualTo1, isGreaterThanOrEqualTo2, isLessThan2, field3IsGreaterThan4));
+    }
+
+
+
+
+    @Test
+    public void shouldReturnNullIfTreeReductionWithFixedValueNotNullInvalidatesTheTree(){
+        AtomicConstraint field3IsGreaterThan4 = new IsGreaterThanConstantConstraint(field3, 4, Collections.emptySet());
+        TreeConstraintNode field2GreaterThanOrEqualTo2Option = new TreeConstraintNode(
+            Collections.singletonList(isGreaterThanOrEqualTo2),
+            Collections.singletonList(
+                new TreeDecisionNode(
+                    new TreeConstraintNode(field3IsGreaterThan4) //something that isn't yet fixed and itself wouldn't be removed
+                )
+            ));
+        DecisionNode decision1 = new TreeDecisionNode(
+            field2GreaterThanOrEqualTo2Option, //field2 > 2
+            new TreeConstraintNode(isLessThan2) //field3
+        );
+
+
+        IsNullConstraint isNull = new IsNullConstraint(field3, Collections.emptySet());
+        AtomicConstraint isNotNull = isNull.negate();
+
+        ConstraintNode tree = new TreeConstraintNode(
+            Collections.singletonList(isNotNull),
+            Collections.singletonList(decision1));
+        ReductiveDecisionTreeReducer reducer = new ReductiveDecisionTreeReducer(
+            fieldSpecFactory,
+            fieldSpecMerger,
+            simplifier);
+        FixedField field3FixedTo3 = new FixedField(field3, Stream.of(3), FieldSpec.Empty, nullMonitor);
+        ReductiveState field3Fixed = initialState.with(field3FixedTo3);
+        field3FixedTo3.getStream().iterator().next(); //move to the first value for field3, i.e. null
+
+        ReductiveConstraintNode reduced = reducer.reduce(tree, field3Fixed);
+
+        //fixed value of field3 of null contradicts the constraint field3 must not be null
+
+        Assert.assertThat(reduced, equalTo(null));
+    }
+
+
+
+    @Test
+    public void shouldReturnNullIfTreeReductionWithFixedValueNullInvalidatesTheTree(){
+        AtomicConstraint field3IsGreaterThan4 = new IsGreaterThanConstantConstraint(field3, 4, Collections.emptySet());
+        TreeConstraintNode field2GreaterThanOrEqualTo2Option = new TreeConstraintNode(
+            Collections.singletonList(isGreaterThanOrEqualTo2),
+            Collections.singletonList(
+                new TreeDecisionNode(
+                    new TreeConstraintNode(field3IsGreaterThan4) //something that isn't yet fixed and itself wouldn't be removed
+                )
+            ));
+        DecisionNode decision1 = new TreeDecisionNode(
+            field2GreaterThanOrEqualTo2Option, //field2 > 2
```

Comment:
As above, update the comment

```suggestion
            field2GreaterThanOrEqualTo2Option, //field2 >= 2
```

Reply:
thanks, changed.